### PR TITLE
add code style and conventions

### LINKED ISSUE #16: Code Style and conventions
**What does the current documentation state?**
The current documentation doesn't provide the code style and conventions to be followed for different repositories.

**Why do you want to improve the statement?**
We can improve this by adding some style conventions and tools that the contributor can enforce to ensure maintenance of the code in the long run. 

Currently only for C++, Python

### PR DESCRIPTION
**Pull request by Tlazypanda**
# Pull Request Template

Fixes #16 

## Description

Added code style and conventions for C++ and Python repositories. 

- [x] This change requires a documentation update

### CODE REVIEW

File: docs/contributing/code_contribution_best_practices.rst [hunk 1]
```diff
@@ -30,6 +30,20 @@ Depending on the language of the repository you are working on, we have an array
...
+- **C++ Repositories**:
+
+  moja global follows the C++ style guide developed by Google for their open-source projects. The google style guide is aimed at enable coders to utilise the power of C++ while at the same time managing the potential complexity that can arise when coding in C++.
+
+  The style guide can be found at: https://google.github.io/styleguide/cppguide.html
+
+  It is possible that exceptions to the google style guide may be specified, in which case they will be listed here. Currently there are no exceptions.
+
+  - Coding enforcement: We know that the style guides are long and detailed and not always easy to adhere to. As such, the intention is to use `Clang-Tidy <http://clang.llvm.org/extra/clang-tidy/>`_ as a tool to check and correct code formatting as determined by the Google C++ style guide. This will be implemented as an automated check through the Continuous Integration system.
+
```
Comment (gmajan) [inline docs/contributing/code_contribution_best_practices.rst:42]: Can you pls add an extra linebreak here? thx (ha ha, this is my only remark.)

File: docs/contributing/code_contribution_best_practices.rst [hunk 2]
```diff
@@ -30,6 +30,20 @@ Depending on the language of the repository you are working on, we have an array
...
+- **Python Repositories**:
+
+  moja global follows the Python Style guide PEP8 that provides coding conventions for Python code. It is fairly common for Python code to follow this style guide. It’s a great place to start since it’s already well-defined.
+
+  This guide can be found at : https://www.python.org/dev/peps/pep-0008/
+
+  - Coding enforcement: In order to enforce the PEP8 conventions along with error detection, `Pylint <https://www.pylint.org/>`_ as a tool can be used and can be integrated with your editor as well.This will be implemented as an automated check through the Continuous Integration system.
```
Comment (gmajan) [inline docs/contributing/code_contribution_best_practices.rst:49]: Space missing before "this"

**Comment by gmajan** [comment]
Sneha, I left 2 small comments. 
Patama, once you have reviewed, pls merge this addition pls.
Or we can discuss on Thrusday how we can coordinate our reviews.

**Comment by Tlazypanda** [comment]
@gmajan @Patamap Have made the changes :sweat_smile: can you review? Thanks :v:

**Review by gmajan — APPROVED** [review]

**Pull request merged**